### PR TITLE
Use standard key events for Notebook keybindings

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
For https://github.com/microsoft/azuredatastudio/issues/17878

I have no idea how this worked in the first place - per https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key 

`The KeyboardEvent interface's key read-only property returns the value of the key pressed by the user, taking into consideration the state of modifier keys such as Shift as well as the keyboard locale and layout.`

So the value should have been `H` instead of `h`

But apparently this works fine on Macs as it is so maybe there's a bug in Chrome there. 

Regardless this isn't a safe way to check these - using the StandardKeyboardEvent should be the preferred way of handling any keyboard related input. 

Note that this still isn't really how we should be handling these things - using our own key handling logic like this is much more error prone and doesn't allow the user to modify the keybindings used for these actions. We should instead be registering commands (with the appropriate when clauses) and using those instead. But such a change would be a riskier thing so for the time being just fixing this logic for the December release.

### CODE REVIEW

**Review by barbaravaldez — APPROVED** [review]

**Review by alanrenmsft — APPROVED** [review]

File: src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts [hunk 1]
```diff
@@ -39,24 +41,25 @@ export class MarkdownToolbarComponent extends AngularDisposable {
...
 				// Italicize text
 				DOM.EventHelper.stop(e, true);
 				await markdownTextTransformer.transformText(MarkdownButtonType.ITALIC);
-			} else if ((e.ctrlKey || e.metaKey) && e.key === 'u') {
+			} else if ((keyEvent.ctrlKey || keyEvent.metaKey) && keyEvent.keyCode === KeyCode.KEY_U) {
 				// Underline text
 				DOM.EventHelper.stop(e, true);
 				await markdownTextTransformer.transformText(MarkdownButtonType.UNDERLINE);
-			} else if ((e.ctrlKey || e.metaKey) && e.shiftKey && e.key === 'k') {
+			} else if ((keyEvent.ctrlKey || keyEvent.metaKey) && e.shiftKey && keyEvent.keyCode === KeyCode.KEY_K) {
```
Comment (lucyzhang929) [inline src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts:58]: Does `e.shiftKey` also need to be changed to `keyEvent.shiftKey`?

File: src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts [hunk 2]
```diff
@@ -39,24 +41,25 @@ export class MarkdownToolbarComponent extends AngularDisposable {
...
 				// Code Block
 				DOM.EventHelper.stop(e, true);
 				await markdownTextTransformer.transformText(MarkdownButtonType.CODE);
-			} else if ((e.ctrlKey || e.metaKey) && e.shiftKey && e.key === 'h') {
+			} else if ((keyEvent.ctrlKey || keyEvent.metaKey) && e.shiftKey && keyEvent.keyCode === KeyCode.KEY_H) {
```
Comment (lucyzhang929) [inline src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts:62]: `keyEvent.shiftKey`

**Review by VasuBhog — APPROVED** [review]

File: src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts [hunk 3]
```diff
@@ -72,33 +74,34 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 	onkeydown(e: KeyboardEvent) {
 		if (DOM.getActiveElement() === this.output?.nativeElement && this.isActive() && this.cellModel?.currentMode === CellEditModes.WYSIWYG) {
 			// Select all text
-			if ((e.ctrlKey || e.metaKey) && e.key === 'a') {
+			const keyEvent = new StandardKeyboardEvent(e);
+			if ((keyEvent.ctrlKey || keyEvent.metaKey) && keyEvent.keyCode === KeyCode.KEY_A) {
 				preventDefaultAndExecCommand(e, 'selectAll');
-			} else if ((e.metaKey && e.shiftKey && e.key === 'z') || (e.ctrlKey && e.key === 'y') && !this.markdownMode) {
+			} else if ((keyEvent.metaKey && keyEvent.shiftKey && keyEvent.keyCode === KeyCode.KEY_Z) || (e.ctrlKey && keyEvent.keyCode === KeyCode.KEY_Y) && !this.markdownMode) {
```
Comment (lucyzhang929) [inline src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts:80]: `keyEvent.ctrlKey` here and the other instances in this file

**Review by lucyzhang929 — APPROVED** [review]

File: src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts:58]: Ah dangit...

**Pull request merged**